Packages/Core/Editor/Utils/AtomNameUtils.cs

```diff
@@ -0,0 +1,35 @@
+using System.Text.RegularExpressions;
+using UnityEditor;
+
+namespace UnityAtoms.Editor
+{
+    public static class AtomNameUtils
+    {
+        public static string FilterLastIndexOf(string result, string identifier)
+        {
+            return result.Contains(identifier) ? result.Substring(result.LastIndexOf(identifier) + 1) : result;
+        }
+
+        public static string CleanPropertyName(string propertyName)
+        {
+            string cleanedProperty = propertyName;
+            if (propertyName[0].ToString() == "_")
+            {
+                cleanedProperty = propertyName.Substring(1);
+            }
+            if (Regex.Match(cleanedProperty, @"[a-zA-Z]").Success)
+            {
+                var index = Regex.Match(cleanedProperty, @"[a-zA-Z]").Index;
+                cleanedProperty = cleanedProperty[index].ToString().ToUpper() + cleanedProperty.Substring(index + 1);
+            }
+            return cleanedProperty;
+        }
+
+        public static string CheckForDuplicateAtom(string atomName)
```

Comment:
It is not really checking for duplicates, but rather giving the atom an unique name based on if there are other assets with the same name. Could we call it `CreateUniqueName` instead?

Reply:
there's utility functions from Unity, [GenerateUniqueAssetPath](https://docs.unity3d.com/ScriptReference/AssetDatabase.GenerateUniqueAssetPath.html), and [GetUniqueName](https://docs.unity3d.com/ScriptReference/ObjectNames.GetUniqueName.html) can be considered for this use case.

Reply:
@fakegood Didn't know about those 2, great tip! However, I believe that in this case the utils that Thimo has made are actually easier to use / maintain.